### PR TITLE
Add googletest framework

### PR DESCRIPTION
**Pull request by natke**
Add googletest ahead of our testing hackathon tomorrow

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -208,3 +208,17 @@ source_group("Sources" FILES ${test_srcs})
...
 
 set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT Tests)
+
+enable_testing()
+
+add_executable(
+  tests
+  test/test.cpp
+)
+target_link_libraries(
```
Comment (baijumeswani) [inline CMakeLists.txt:218]: Do you also want to link to onnxruntime-genai?

Comment (natke) [reply CMakeLists.txt:218]: Probably!

Comment (natke) [reply CMakeLists.txt:224]: Done

**Review by snnn — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (snnn) [reply CMakeLists.txt:224]: It's not enough. But we can do it later, increamentally.

File: CMakeLists.txt [hunk 2]
```diff
@@ -210,21 +210,22 @@ source_group(TREE ${GENERATORS_ROOT} FILES ${generator_srcs})
...
+  PRIVATE
+  ${CMAKE_SOURCE_DIR}/src
 )
 
+
 target_link_libraries(
-  onnxruntime-genai
   unit_tests
+  PRIVATE
   GTest::gtest_main
```
Comment (snnn) [inline CMakeLists.txt:227]: Here we need to link to GenAI static libraries, and all the dependencies that GenAI uses, such as SPM.

Comment (natke) [reply CMakeLists.txt:227]: Is there a variable for that, or do we need to list them out?

Comment (snnn) [reply CMakeLists.txt:227]: No such a thing yet. We can do it later.

**Pull request merged**